includes/integrations/class-sst-deposits-for-wc.php

```diff
@@ -0,0 +1,658 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Deposits for WooCommerce integration for Simple Sales Tax.
+ *
+ * NOTE: Applying a discount on the full cart total including tax
+ * is NOT supported (`deposit_on_total_incl_tax`).
+ *
+ * @author Brett Porcelli <bporcelli@taxcloud.com>
+ */
+class SST_Deposits_For_WC {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// CART HOOKS
+		add_filter(
+			'woocommerce_get_cart_contents',
+			array( $this, 'add_tax_to_totals' )
+		);
+
+		$checkout_deposits_enabled = (
+			'yes' === get_option( 'checkout_deposits_enabled' ) &&
+			'no' !== get_option( 'default_deposits_enabled' )
+		);
+
+		if ( $checkout_deposits_enabled ) {
+			add_filter(
+				'wootax_cart_packages',
+				array( $this, 'filter_cart_packages' )
+			);
+			add_filter(
+				'woocommerce_calculated_total',
+				array( $this, 'set_post_data' )
+			);
+			add_filter(
+				'woocommerce_calculated_total',
+				array( $this, 'fix_cart_deposit_info' ),
+				1110,
+				2
+			);
+			add_filter(
+				'dfw_total_cart_amount_price_figure_html',
+				array( $this, 'fix_total_cart_amount' )
+			);
+		}
+
+		// ORDER HOOKS
+		add_filter(
+			'sst_should_calculate_order_tax',
+			array( $this, 'should_calculate_order_tax' ),
+			10,
+			3
+		);
```

Comment:
May want to drop this in favor of just relying on the `woocommerce_before_order_object_save` hook.

Reply:
f36c2a7